mh/pat.py

```diff
@@ -256,9 +256,13 @@ def recvAnsConnection(self, packet_id, data, seq):
 
         The games sends the PAT environment properties.
         """
-        connection_data = pati.ConnectionData.unpack(data)
-        self.server.debug("Connection: {!r}".format(connection_data))
-        self.sendNtcLogin(5, connection_data, seq)
+        settings = pati.ConnectionData.unpack(data)
+        self.server.debug("Connection: {!r}".format(settings))
+        pat_ticket = settings.pat_ticket if "pat_ticket" in settings else \
+                     settings.online_support_code if "online_support_code" in settings \
+                     else ""
+        self.server.info("Client {} Ticket `{}`".format(self.client_address, pat_ticket))
```

Comment:
```suggestion
        pat_ticket = b""
        if "pat_ticket" in settings:
            _, pat_ticket = pati.unpack_any(settings.pat_ticket)
        elif "online_support_code" in settings:
            _, pat_ticket = pati.unpack_any(settings.online_support_code)
        self.server.info("Client {} Ticket `{}`".format(self.client_address,
                                                        pat_ticket))
```
You need to unpack data from PAT object members, otherwise you'll get bytes describing its item type.

Reply:
lol, I forgot about that